zend/value.cpp

```diff
@@ -203,12 +203,12 @@ Value::Value(const IniValue &value) : Value((const char *)value) {}
  */
 Value::Value(const Value &that)
 {
-    if (Z_ISREF_P(that._val)) {
-        zend_assign_to_variable(_val, that._val, IS_VAR);
-    }
-    else {
-        ZVAL_COPY(_val, that._val);
+    zval* z = that._val;
+    while (Z_ISREF_P(z)) {
+        ZVAL_DEREF(z);
     }
```

Comment:
When copying a variable which is a reference to another variable, we need to get to the original variable and copy it, NOT create a copy of the reference itself. This differs from PHP5, as PHP7 has a dedicated IS_REFERENCE type for references.